src/pudl/extract/epacems.py

```diff
@@ -191,26 +191,32 @@ def get_data_frame(self, partition: EpaCemsPartition) -> pd.DataFrame:
 
     def _csv_to_dataframe(
         self,
-        csv_file: Path,
+        csv_path: Path,
         ignore_cols: dict[str, str],
         rename_dict: dict[str, str],
         dtype_dict: dict[str, type],
+        chunksize: int = 100_000,
     ) -> pd.DataFrame:
         """Convert a CEMS csv file into a :class:`pandas.DataFrame`.
 
         Args:
-            csv (file-like object): data to be read
+            csv_path: Path to CSV file containing data to read.
 
         Returns:
-            A DataFrame containing the contents of the CSV file.
+            A DataFrame containing the filtered and dtyped contents of the CSV file.
         """
-        return pd.read_csv(
-            csv_file,
+        chunk_iter = pd.read_csv(
+            csv_path,
             index_col=False,
             usecols=lambda col: col not in ignore_cols,
             dtype=dtype_dict,
-            low_memory=False,
-        ).rename(columns=rename_dict)
+            chunksize=chunksize,
+            low_memory=True,
```

Comment:
I don't know what it does on the inside, but `low_memory=True` tries to be more memory efficient, and `chunksize` reads in batches of records and processes them one at a time, returning an iterator of  dataframes (one per chunk) rather than a single dataframe.

Reply:
"Internally process the file in chunks, resulting in lower memory use while parsing, but possibly mixed type inference. To ensure no mixed types either set False, or specify the type with the dtype parameter. Note that the entire file is read into a single [DataFrame](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html#pandas.DataFrame) regardless, use the chunksize or iterator parameter to return the data in chunks. (Only valid with C parser)."

Reply:
If all of the `CategoricalDtype()` columns are well-behaved and we actually **do** know what the values will be ahead of time, probably the right thing to do in here is define the categories ahead of time, and then this chunking will use low-memory dtypes that can also be concatenated without being `object`ified.

Reply:
That would make sense, but to be consistent with how we're handling other datasets I'd probably want to map all the column dtypes in `fields.py` and `codes.py`, as there are currently no column by column type enforcements on EPA CEMS data in the same way that we have for EIA/FERC data. This seems out of scope of this current PR and I'm tempted to move it into a separate issue and handle it there.